python-package/lightgbm/engine.py

```diff
@@ -486,12 +492,16 @@ def cv(params, train_set, num_boost_round=100,
     """
     if not isinstance(train_set, Dataset):
         raise TypeError("Training only accepts Dataset object")
-
     params = copy.deepcopy(params)
-    if fobj is not None:
-        for obj_alias in _ConfigAliases.get("objective"):
-            params.pop(obj_alias, None)
-        params['objective'] = 'none'
+    params = _choose_param_value(
+        main_param_name='objective',
+        params=params,
+        default_value=None
```

Comment:
Same question as above for `default_value=None`.

Reply:
this could work the same as before with `default_value='none'`, so that user knows he needs an `object function`. 

One possibility could be that `CVBooster` handles a similar logic as `LGBMModel._process_params`

```python
if self._objective is None:
    if isinstance(self, LGBMRegressor):
        self._objective = "regression"
    elif isinstance(self, LGBMClassifier):
        if self._n_classes > 2:
            self._objective = "multiclass"
        else:
            self._objective = "binary"
    elif isinstance(self, LGBMRanker):
        self._objective = "lambdarank"
    else:
        raise ValueError("Unknown LGBMModel type.")
```